### PR TITLE
Fixes #22119 - use SHA instead of MD5 for digesting

### PR DESCRIPTION
**Pull request by iNecas**
MD5 is disabled in FIPS environments.

### CODE REVIEW

**Comment by iNecas** [comment]
[test katello]

**Comment by iNecas** [comment]
Test failures are related, as https://projects.theforeman.org/issues/23363 counts on using the hexdigest once available. I will try to regenerate the VCR with this change and send a PR to katello (or I will ask katello folks if I fail doing it by myself)

**Review by lzap — APPROVED** [review]
LGTM

**Comment by iNecas** [comment]
After updating VCR cassette, the failing katello test got green as well when running locally, Katello PR should get green after this gets merged https://github.com/Katello/katello/pull/7732

**Comment by iNecas** [comment]
@ares anything to add or good to merge? If it makes you more comfortable, @pondrejk has been doing tests with this change for months and have not observed this breaking anything.

**Comment by ares** [comment]
That definitely helps with confidence, please give me one more day to apply on my setup and do quick tests. I don't expect any problems, but still :-) Thanks

**Review by ares — APPROVED** [review]
for the reference, this is what introduced the support in rails https://github.com/rails/rails/commit/82822a34217503336d51b7baab82cd18cf71e435#diff-44aa07f65459835fba75ab75fe3863efR185

I didn't find anything broken, thanks @iNecas, merging!

**Pull request merged**
